src/main.js

```diff
@@ -1,96 +1,148 @@
-import logger from '@financial-times/n-logger';
-
-const MASK_SEQUENCE = '*****';
-
-export default class {
-
-	constructor (sensitiveFields = []) {
-		const fieldsToMask = sensitiveFields.join('|');
-		this.sensitiveFields = new RegExp(`(${fieldsToMask})[\\\s]*\\=[\\\s]*[\\\]?[\\"\\']?([\\S]+)[\\\]?[\\"\\']?|(${fieldsToMask})`, 'ig');
+const logger = require('@financial-times/n-logger').default;
+
+/**
+ * MaskLogger Class
+ * @example
+ * const logger = new MaskLogger(['denyField'], ['allowField'], '*MASKED*');
+ * logger.debug('Data which may contain sensitive information', data);
+ */
+class MaskLogger {
+	/**
+	 * Create a MaskLogger
+	 * @param {array<string>} denyList Extra field names to mask
+	 * @param {array<string>} allowList Field names to allow from the default deny list
+	 * @param {string?} maskString Mask to be applied to sensitive values
+	 */
+	constructor (denyList = [], allowList = [], maskString = '*****') {
+		this.maskString = maskString;
+		this.maskList = [
+			'email',
+			'password',
+			'name',
+			'firstName',
+			'lastName',
+			'phone',
+			'primaryTelephone',
+			'postcode',
+			'session',
```

Comment:
@Financial-Times/ft-com-conversion any default denied fields I've missed?

Reply:
I would try to prevent leaking of keys, that happened in Splunk/Sentry due to log of the headers. Going back to my previous comment I think we should have `key` and that should work with stuff like `ft-backend-key` or similar. I think is better that we mask too much rather than less. Then people can whitelist if they want.

Reply:
Again I feel if we make this too greedy then it will mask things we may not want it too. `key` can be found in lots of other words.